src/hit/api/evaluator/homomorphic.cpp

```diff
@@ -238,6 +259,19 @@ namespace hit {
         return backend_encryptor->object;
     }
 
+    Bootstrapper &HomomorphicEval::get_bootstrapper() {
+        if (!(context->btp_params.has_value())) {
+            LOG_AND_THROW_STREAM("CKKS parameters do not specify bootstrapping parameters.");
+        }
+
+        if (backend_bootstrapper.get() == nullptr || backend_bootstrapper->params != context->params) {
+            ParameterizedLattigoType<Bootstrapper> *tmp = new ParameterizedLattigoType<Bootstrapper>(
+                newBootstrapper(context->params, context->btp_params.value(), btp_keys), context->params);
+            backend_bootstrapper.reset(tmp);
+        }
```

Comment:
Nit: will tmp get freed somewhere? AddressSanitizer may need to get set up in CI when there is time.

Reply:
This is handled by `thread_specific_ptr`. Once `reset()` is called, the smart pointer takes control of the object and its life cycle. The idea is that it will automatically clean up the object when the thread exits (which the programmer may not have much visibility into). Docs [here](https://www.boost.org/doc/libs/1_35_0/doc/html/thread/thread_local_storage.html):
> When a thread exits, the objects associated with each boost::thread_specific_ptr instance are destroyed. By default, the object pointed to by a pointer p is destroyed by invoking delete p

Reply:
Greg and I determined that my reasoning is not correct, and that there is a bug somewhere that resulted in me needing to have the second part of the condition. However, it works for now, so I'm going to merge this PR and open an issue to track the bug, see #146 .